esphome/components/lvgl/rotary_encoders.py

```diff
@@ -0,0 +1,62 @@
+import esphome.codegen as cg
+from esphome.components.binary_sensor import BinarySensor
+from esphome.components.rotary_encoder.sensor import RotaryEncoderSensor
+import esphome.config_validation as cv
+from esphome.const import CONF_GROUP, CONF_ID, CONF_SENSOR
+
+from .defines import (
+    CONF_ENTER_BUTTON,
+    CONF_LEFT_BUTTON,
+    CONF_LONG_PRESS_REPEAT_TIME,
+    CONF_LONG_PRESS_TIME,
+    CONF_RIGHT_BUTTON,
+    CONF_ROTARY_ENCODERS,
+)
+from .helpers import lvgl_components_required
+from .lvcode import add_group, lv, lv_add, lv_expr
+from .schemas import ENCODER_SCHEMA
+from .types import lv_indev_type_t
+
+ROTARY_ENCODER_CONFIG = cv.ensure_list(
+    ENCODER_SCHEMA.extend(
+        {
+            cv.Required(CONF_ENTER_BUTTON): cv.use_id(BinarySensor),
+            cv.Required(CONF_SENSOR): cv.Any(
+                cv.use_id(RotaryEncoderSensor),
+                cv.Schema(
+                    {
+                        cv.Optional(CONF_LEFT_BUTTON): cv.use_id(BinarySensor),
+                        cv.Optional(CONF_RIGHT_BUTTON): cv.use_id(BinarySensor),
+                    }
+                ),
+            ),
+        }
+    )
+)
+
+
+async def rotary_encoders_to_code(var, config):
+    for enc_conf in config.get(CONF_ROTARY_ENCODERS, ()):
+        lvgl_components_required.add("KEY_LISTENER")
+        lvgl_components_required.add("ROTARY_ENCODER")
+        lpt = enc_conf[CONF_LONG_PRESS_TIME].total_milliseconds & 0xFFFF
+        lprt = enc_conf[CONF_LONG_PRESS_REPEAT_TIME].total_milliseconds & 0xFFFF
```

Comment:
Not needed with the max validation
```suggestion
        lpt = enc_conf[CONF_LONG_PRESS_TIME].total_milliseconds
        lprt = enc_conf[CONF_LONG_PRESS_REPEAT_TIME].total_milliseconds
```